macros/core/PGauxiliaryFunctions.pl

```diff
@@ -555,5 +556,32 @@ sub random_subset {
 	return wantarray ? @out : \@out;
 }
 
+=head2 repeated function
+
+Usage: C<repeated(@list)>
+
+This function returns a list of every repeated elements in @list. Comparison is made
+using ==, so two elements may be considered 'repeated' even when they are not literally
+equal.
+
+Note that the function will return () if there are no repeated elements, which
+is false as a boolean. So !repeated(@list) is a way to check if the list has no repeated
+values.
+
+Also note that generally if two items are equivalent, both will be in the returned list.
+However occasionally with MathObjects, x == y is true while y == x is false, and then
+only x (the one that makes the relation true when it is on the left) will be included
+in the returned list.
+
+=cut
+
+sub repeated {
+	my @return;
+	for my $x (@_) {
+		push(@return,$x)  if (grep { $_ == $x } (@_)) > 1;
```

Comment:
This is now failing perltidy tests here, there is a misplaced space: `push(@return, $x) if (grep { $_ == $x } (@_)) > 1;`.